src/ORM/FieldType/DBForeignKey.php

```diff
@@ -84,12 +84,21 @@ public function scaffoldFormField($title = null, $params = null)
 
         // Add the count of the list to a cache for subsequent calls
         if (!isset(static::$foreignListCache[$hasOneClass])) {
+            // Let the DB do the threshold check as it will be faster - depending on the SQL engine it might only have
+            // to count indices
+            $dataQuery = $list->dataQuery()->getFinalisedQuery();
+            $dataQuery->setDistinct(false);
+            $dataQuery->setOrderBy(false);
+            $dataQuery->setSelect(['over_threshold' => 'count(*) > ' . (int) $threshold]);
+            $result = $dataQuery->execute()->column('over_threshold')[0];
             static::$foreignListCache[$hasOneClass] = [
-                'count' => $list->count(),
+                'overThreshold' => is_bool($result) ? $result : ($result === '1' || $result === 't'),
```

Comment:
Could you break these changes out into a separate commit?

Reply:
I'm going to raise this as a patch PR separately.